its/src/test/java/com/sonar/it/scanner/msbuild/sonarcloud/IncrementalPRAnalysisTest.java

```diff
@@ -61,15 +62,18 @@ void master_emptyCache() throws IOException {
 
   @Test
   void prWithoutChanges_producesUnchangedFilesWithAllFiles() throws IOException {
-    var projectDir = TestUtils.projectDir(basePath, PROJECT_NAME);
+    var context = AnalysisContext.forCloud(SONARCLOUD_PROJECT_KEY, basePath, PROJECT_NAME);
+    context.runAnalysis();  // Initial build - master.
 
-    CloudUtils.runAnalysis(projectDir, SONARCLOUD_PROJECT_KEY); // Initial build - master.
-    var logs = CloudUtils.runAnalysis(projectDir, SONARCLOUD_PROJECT_KEY, prArguments); // PR analysis.
+    for (var property : prArguments) {
+      context.begin.setProperty(property.name(), property.value());
+    }
```

Comment:
Can we have an overload/method `context.begin.setProperties(prArguments)`? Maybe using Java varargs.

Reply:
We can. But as this is the only use, I'd leave it like this for now.

Reply:
#2416